accepted/00070-akka.md

```diff
@@ -0,0 +1,134 @@
+ - error handling was not tested thoroughly, might need rework
+ - the Cucumber integration testsuite was not run on the resulting code
+ - performance tests will have to be conducted to make sure there is no regression (and evaluate possible speedups)
+ - the reflections-based mechanism to find out workers will have to be considered and it might be removed in exchange for an explicit approach
+ - most but not all Actor Messages were made `Serializable`, more refactoring might be needed to ensure this is always the case
+ - Akka tuning knobs were not all exposed, or exposed by reusing existing configuration variables. New variables might need to be introduced and properly documented
+ - the `DeferredActor` implementation is at the moment not completely written in functional style
+
+5. move specific Taskomatic task facilities (eg. `ErrataCacheDriver`, `SSHPushDriver`) to Akka
+6. refactor Taskomatic itself to use Akka
+
+Last point is about having Quartz-based Taskomatic logic only caring aobut the triggering of jobs, and leave to Akka their execution. Basically when Taskomatic fires (eg. on cron line) it will send a message to an Akka actor which would then carry out the actual computation.
+
+## Following Steps
+
+Following points need further research when points 1.-6. are completed, and possibly need follow-up RFCs, still are reported here for completeness.
+
+### Move Taskomatic code in the Tomcat web app
+
+After all asynchronous computation is moved to Akka, it might make sense to consider moving all Taskomatic code out of the currently separated daemon snd into the main Tomcat web app.
+
+Architecturally, such a change would have pros and cons:
+ - pro: we could remove the internal Tomcat XMLRPC API, eliminating substantial complexity and the synchronization/availability problems we currently face (eg. [the race condition in `MinionActionExecutor`](https://github.com/uyuni-project/uyuni/blob/4.1-alpha1/java/code/src/com/redhat/rhn/taskomatic/task/MinionActionExecutor.java#L82-L102), as per [downstream issue 10657](https://github.com/SUSE/spacewalk/issues/10657))
+ - pro: with one DB connection pool and one Akka thread pool, it should result in an easier-to-tune Server
+ - pro: currently many support issues boil down to undetected Taskomatic failures. With one only web app, it should be easier to surface errors in a more user-visible way
+ - con: unexpected Taskomatic errors may bring Tomcat down as well. At the moment this does not happen, but as we rely on Taskomatic much more than in original Spacewalk, in many cases we already end up with a non-functioning Server anyway
+
+### Leverage Akka Clustering to offload computation onto other machines
+
+Akka has an optional module called Clustering that allows to transparently move computation across machines, specifically by instantiating Actors on remote nodes and letting Akka route messages to them.
```

Comment:
makes sense to refer it allows across different machines or different processes on the same machine? For example, we can keep just one machine as we have now, move taskmatic event firing to inside tomcat, but the execution could be done in a worker in the same machine.